testing_utilities.go

```diff
@@ -0,0 +1,359 @@
+	assert.Equal(t, str1, str2, "Expected strings to match")
+}
+
+func compareAnnotations(t *testing.T, reader1 ion.Reader, reader2 ion.Reader) {
+	// TODO: Add SymbolToken logic here once SymbolTokens are available
+
+	assert.Equal(t, reader1.Annotations(), reader2.Annotations(), "Expected symbol sequences to match")
+}
+
+func compareAnnotationSymbols(t *testing.T, reader1, reader2 ion.Reader) {
+	// TODO: Add SymbolToken logic here once SymbolTokens are available
+}
+
+func compareHasAnnotations(t *testing.T, reader1, reader2 ion.Reader) {
+	// TODO: Add SymbolToken logic here once SymbolTokens are available
+}
+
+func compareScalars(t *testing.T, ionType ion.Type, isNull bool, reader1 ion.Reader, reader2 ion.Reader) {
+	switch ionType {
+	case ion.BoolType:
+		if !isNull {
+			value1, err := reader1.BoolValue()
+			assert.NoError(t, err, "Something went wrong executing reader1.BoolValue()")
+
+			value2, err := reader2.BoolValue()
+			assert.NoError(t, err, "Something went wrong executing reader2.BoolValue()")
+
+			assert.Equal(t, value1, value2, "Expected bool values to match")
+		}
+	case ion.IntType:
+		if !isNull {
+			intSize, err := reader1.IntSize()
+			assert.NoError(t, err, "Something went wrong executing reader1.IntSize()")
+
+			switch intSize {
+			case ion.Int32, ion.Int64:
+				int1, err := reader1.Int64Value()
+				assert.NoError(t, err, "Something went wrong executing reader1.Int64Value()")
+
+				int2, err := reader2.Int64Value()
+				assert.NoError(t, err, "Something went wrong executing reader2.Int64Value()")
+
+				assert.Equal(t, int1, int2, "Expected int values to match")
+			case ion.Uint64:
+				uint1, err := reader1.Uint64Value()
+				assert.NoError(t, err, "Something went wrong executing reader1.Uint64Value()")
+
+				uint2, err := reader2.Uint64Value()
+				assert.NoError(t, err, "Something went wrong executing reader2.Uint64Value()")
+
+				assert.Equal(t, uint1, uint2, "Expected uint values to match")
+			case ion.BigInt:
+				bigInt1, err := reader1.BigIntValue()
+				assert.NoError(t, err, "Something went wrong executing reader1.BigIntValue()")
+
+				bigInt2, err := reader2.BigIntValue()
+				assert.NoError(t, err, "Something went wrong executing reader2.BigIntValue()")
+
+				assert.Equal(t, bigInt1, bigInt2, "Expected big int values to match")
+			default:
+				t.Error("Expected intSize to be one of Int32, Int64, Uint64, or BigInt")
+			}
+		}
+	case ion.FloatType:
+		if !isNull {
+			float1, err := reader1.FloatValue()
+			assert.NoError(t, err, "Something went wrong executing reader1.FloatValue()")
+
+			float2, err := reader2.FloatValue()
+			assert.NoError(t, err, "Something went wrong executing reader2.FloatValue()")
+
+			if math.IsNaN(float1) && math.IsNaN(float2) {
+				assert.Equal(t, float1, float2, "Expected NaN float values to match")
+			} else if math.IsNaN(float1) || math.IsNaN(float2) {
+				assert.NotEqual(t, float1, float2, "Expected float values to differ")
```

Comment:
Message for this assert is confusing. I would propose, "IsNan value should differ from a non-IsNaN value"

Reply:
Updated message.